src/dashboardUsers/controller/get-userByAdmin.js

```diff
@@ -1,39 +1,32 @@
-require('../../authentication/controller/passport-config');
-
-const UserModel = require('../../authentication/model/user-model');
-
-
-
-exports.getUser =() => async(req,res,next) => {
-         const user = {};
-         if(!req.body.email){
-            return res.status(400).json({
-            error: 'Sorry, but email is required!',
-            data: []
-            });
-         }
-         const {email} = req.body;
-         const foundUser = await UserModel.findOne({ email });
-         
-         if (!foundUser) {
-             return res.status(400).json({
-                error: 'Sorry, but the user cannot be found!',
-                data: []
-             });
-         } else {
-
-        user.firstname=foundUser.firstname,
-        user.lastname=foundUser.lastname, 
-        user.email=foundUser.email,
-        user.phone=foundUser.phone,
-        user.role=foundUser.role,
-
-        req.user = user;
-
-        next();
-         }  
-       
-}
-
-
-
+const UserModel = require("../../authentication/model/user-model");
+
+exports.getUser = () => async (req, res, next) => {
+  const { email } = req.body;
+  const user = {};
+
+  if (!email) {
+    return res.status(400).json({
+      error: "Sorry, but email is required!",
+      data: [],
+    });
+  }
+  const foundUser = await (await UserModel.findOne({ email })).toObject();
```

Comment:
Here I tested the toObject () function to turn the foundUser result into a pure object with no extra data. the first time it waits to find the user, and after that it waits again to have the result transformed by the toObject () function. Otherwise i recieve an error.